numpy/core/src/umath/loops.c.src

```diff
@@ -1644,6 +1644,41 @@ TIMEDELTA_mm_q_floor_divide(char **args, npy_intp *dimensions, npy_intp *steps,
     }
 }
 
+NPY_NO_EXPORT void
+TIMEDELTA_mm_qm_divmod(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    BINARY_LOOP_TWO_OUT {
+        const npy_timedelta in1 = *(npy_timedelta *)ip1;
+        const npy_timedelta in2 = *(npy_timedelta *)ip2;
+        if (in1 == NPY_DATETIME_NAT || in2 == NPY_DATETIME_NAT) {
+            npy_set_floatstatus_invalid();
+            *((npy_timedelta *)op1) = 0;
+            *((npy_timedelta *)op2) = NPY_DATETIME_NAT;
+        }
+        else if (in2 == 0) {
+            npy_set_floatstatus_divbyzero();
+            *((npy_timedelta *)op1) = 0;
+            *((npy_timedelta *)op2) = 0;
+        }
+        else {
+            const npy_timedelta rem = in1 % in2;
+            if ((in1 > 0) == (in2 > 0) || rem == 0) {
+                *((npy_timedelta *)op2) = rem;
+            }
+            else {
+                *((npy_timedelta *)op2) = rem + in2;
+            }
+
+            if (((in1 > 0) != (in2 > 0)) && (in1 % in2 != 0)) {
```

Comment:
This condition is the same as the one above - can you combine the ifs?

Reply:
ok, I've simplified the conditionals